packages/editor/src/components/browser/TypeBrowser.tsx

```diff
@@ -46,7 +55,16 @@ const TypeBrowser = ({ value, onChange, onDoubleClick, initSearchFilter }: TypeB
   const [showHelper, setShowHelper] = useState(false);
 
   useEffect(() => {
-    const typeComparator = (a: TypeBrowserObject, b: TypeBrowserObject) => {
+    const typeComparator = (a: TypeBrowserObject, b: TypeBrowserObject, ivyType?: boolean) => {
+      if (ivyType) {
+        const aHasJava = a.fullQualifiedName.includes('java.');
+        const bHasJava = b.fullQualifiedName.includes('java.');
+        if (aHasJava && !bHasJava) {
+          return -1;
+        } else if (!aHasJava && bHasJava) {
+          return 1;
+        }
+      }

```

Comment:
Never do this! Deduplicate your code is good, but with this additional parameter you change the whole function, so create a new function for this case, because it has nothing do to with the other one.

If you write an if with a return state, you can also write the else if as if, especially if you don't write an else case.

```ts
const ivyTypeComparator = (a: TypeBrowserObject, b: TypeBrowserObject) => {
  const aHasJava = a.fullQualifiedName.includes('java.');
  const bHasJava = b.fullQualifiedName.includes('java.');
  if (aHasJava && !bHasJava) {
    return -1;
  }
  if (!aHasJava && bHasJava) {
    return 1;
  }
  return typeComparator(a, b);
}
```

Reply:
And are you shure that 'java.' is the best to discover ivy types?
I can imagine that someone else also writes a package name with 'java.' in it or that our types have 'ivy' in it or something like that

Reply:
Thanks, from my point of view this should not be a problem as this (ivy)typeComparator is currently only used for the ivyTypes array and this remains static.